contrib/src/templates/context.rs

```diff
@@ -11,12 +11,15 @@ pub struct Context {
     pub root: PathBuf,
     /// Mapping from template name to its information.
     pub templates: HashMap<String, TemplateInfo>,
-    /// Mapping from template name to its information.
-    pub engines: Engines
+    /// Loaded template engines
+    pub engines: Engines,
 }
 
 impl Context {
-    pub fn initialize(root: PathBuf) -> Option<Context> {
+    pub fn initialize(
+        root: PathBuf,
+        customize_callback: &(Fn(&mut Engines) + Send + Sync + 'static),
```

Comment:
Would be cleaner to remove the Send+Sync+'static bounds here, it's only at the level of the Fairing that these bounds are necessary